.gitbook/0-about-panacea/1-data-exchange-protocol/2-How-DEP-works.md

```diff
@@ -0,0 +1,69 @@
+In this process, the provider encrypts the data with the public key of the oracle and its own private key.
+The oracle private key is sealed in the enclave environment, so the data cannot be decrypted by anyone except the oracle.
+
+### Data privacy in oracle
+
+In the process of [Data Validation](../../3-protocol-devs/1-dep-specs/4-data-validation.md), the oracle decrypts the encrypted data with oracle private key and obtains the original data.
+However, the oracle we defined must run in an enclave environment, and the enclave memory cannot be read or written outside the enclave.
+Therefore, it is impossible to leak the original data and data privacy is continuously protected.
+
+### Data privacy between provider & consumer
+
+After completing the [Data Validation](../../3-protocol-devs/1-dep-specs/4-data-validation.md), oracle encrypts the data and provides this encrypted to the consumer.
+t this time, a symmetric key based on the oracle private key is generated, so the data cannot be decrypted until the consumer also receives this symmetric key.
+After the [Incentive Distribution](../../3-protocol-devs/1-dep-specs/6-incentives.md) process is completed, the consumer can request the symmetric key from the oracle.
+The oracle checks whether the incentive distribution process is completed from the panacea blockchain, and provides the secret key encrypted based on the consumer's public key.
+As a result, it is impossible for the provider to obtain the original data before the incentive is paid, and the secret key cannot be obtained except by the consumer.
+
+## For data validation
+
+Oracle validates the data after receiving encrypted data from the provider.
+After the basic deal status, data hash validation, oracle checks whether the data satisfies the requirements specified in the deal.
+here are two types of data requirements that can be specified in the deal (json schema validation, presentation definition validation).
+
+[Json schema](https://json-schema.org) validation checks whether the data satisfies the conditions for the schema definition.
+[Presentation definition](https://identity.foundation/presentation-exchange/#presentation-definition) validation checks that the data is in the form of a did-based verifiable presentation and satisfies the presentation definition.
+This validation can additionally verify that the data was created from a verifiable credential issued by a Certificate Authority, and thus can verify that the data is authenticated by the Certificate Authority.
+
+## Atomic incentive distribution
+
+After receiving the certificate from oracle, the provider can get the incentive through [Incentive Distribution](../../3-protocol-devs/1-dep-specs/6-incentives.md) process.
+In this process, panacea blockchain checks the certificate information and immediately completes the incentive payment to the provider, and at the same time enables the consumer to obtain the secret key to decrypt the provided data.
```

Comment:
Perhaps we should use the term consent instead of certificate.

Reply:
I fixed it in [ac61010](https://github.com/medibloc/panacea-core/pull/648/commits/ac610105fe46dc3198e0218f297b72d4676d0151).